tests/dist/mpi/test_migration.cpp

```diff
@@ -28,15 +27,80 @@ TEST_CASE_METHOD(DistTestsFixture, "Test migrating an MPI execution", "[mpi]")
     msg.set_ismpi(true);
     msg.set_mpiworldsize(mpiWorldSize);
     msg.set_recordexecgraph(true);
+    // Set a low migration check period to detect the mgiration right away
     msg.set_migrationcheckperiod(migrationCheckPeriod);
-    msg.set_cmdline(std::to_string(migrationCheckPeriod));
+    int numLoops = 10000;
+    // Try to migrate at 50% of execution
+    int checkAt = 5;
+    msg.set_cmdline(fmt::format("{} {}", checkAt, numLoops));
 
     // Call the functions
     sch.callFunctions(req);
 
+    // Sleep for a while to let the scheduler schedule the MPI calls, and then
+    // update the local slots so that a migration opportunity appears
+    SLEEP_MS(500);
```

Comment:
The request we are sending in the `sch.callFunctions` line is a request for an MPI application. Thus it is just one message, with the field `msg.mpiworldsize()` set to the `worldSize`.

Once the request is scheduled and has started executing, we will call `MpiWorld::create` that will then issue a _different_ call to `sch.CallFunctions` with `worldSize - 1` functions.

The way we make a migration opportunity appear in this test is by manually modifying the resources available in the main host by calling `sch.setThisHostResources`. On a high level, we do:

```
# Set low resource availability

# Call function with one function (will internally call the method another time)
sch.callFunctions

# Set high resource availability so that a migration opportunity appears
```

It is important than both the explicit (and the implicit) call to `sch.callFunctions` see the low resource availability. Hence the sleep.